File: holoken-app/src/main/java/com/holokenmod/ui/grid/GridCellSizeService.java
package com.holokenmod.ui.grid;

public class GridCellSizeService {
	private static final GridCellSizeService INSTANCE = new GridCellSizeService();
	
	private int cellSizePercent = 100;
	private GridCellSizeListener listener = null;
	
	public static GridCellSizeService getInstance() {
		return INSTANCE;
	}
	
	public int getCellSizePercent() {
		return cellSizePercent;
	}
	
	public void setCellSizePercent(int cellSizePercent) {
		this.cellSizePercent = cellSizePercent;
		
		if (listener != null) {
			listener.cellSizeChanged(cellSizePercent);
		}
	}
	
	public void setCellSizeListener(GridCellSizeListener listener) {
		this.listener = listener;
	}
}

File: holoken-app/src/main/java/com/holokenmod/ui/grid/GridCellSizeService.kt
package com.holokenmod.ui.grid

class GridCellSizeService {
    var cellSizePercent = 100
        set(cellSizePercent) {
            field = cellSizePercent
            listener?.cellSizeChanged(cellSizePercent)
        }

    private var listener: GridCellSizeListener? = null

    fun setCellSizeListener(listener: GridCellSizeListener?) {
        this.listener = listener
    }
}